awx/ui_next/src/screens/Inventory/InventoryDetail/InventoryDetail.jsx

```diff
@@ -11,37 +11,28 @@ import DeleteButton from '@components/DeleteButton';
 import ContentError from '@components/ContentError';
 import ContentLoading from '@components/ContentLoading';
 import { InventoriesAPI } from '@api';
+import useRequest from '@util/useRequest';
 import { Inventory } from '../../../types';
 
 function InventoryDetail({ inventory, i18n }) {
-  const [instanceGroups, setInstanceGroups] = useState([]);
-  const [hasContentLoading, setHasContentLoading] = useState(true);
-  const [contentError, setContentError] = useState(null);
   const history = useHistory();
-  const isMounted = useRef(null);
+
+  const {
+    result: instanceGroups,
+    isLoading,
+    error,
+    request: fetchInstanceGroups,
+  } = useRequest(
+    useCallback(async () => {
+      const { data } = await InventoriesAPI.readInstanceGroups(inventory.id);
+      return data.results;
+    }, [inventory.id]),
+    []
+  );
 
   useEffect(() => {
-    isMounted.current = true;
-    (async () => {
-      setHasContentLoading(true);
-      try {
-        const { data } = await InventoriesAPI.readInstanceGroups(inventory.id);
-        if (!isMounted.current) {
-          return;
-        }
-        setInstanceGroups(data.results);
-      } catch (err) {
-        setContentError(err);
-      } finally {
-        if (isMounted.current) {
-          setHasContentLoading(false);
-        }
-      }
-    })();
-    return () => {
-      isMounted.current = false;
-    };
-  }, [inventory.id]);
+    fetchInstanceGroups(inventory.id);
```

Comment:
At the top where we define the request hook, we pass the id value from the inventory prop to the api method, so it doesn't seem like passing `inventory.id` on this line will do anything. I don't see where we're using _this_ argument `fetchInstanceGroups(inventory.id)` in the code.